From 8a5cdc5e45a3f19c17a97a6d3f65c3ea5aa7eeb6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Wed, 15 Feb 2023 12:40:39 +0100
Subject: [PATCH] frontend/dri: Initialize callbacks in
 dri_swrast_kms_init_screen

This was missed in the commit below.

Closes: https://gitlab.freedesktop.org/mesa/mesa/-/issues/8232
Fixes: 7d5b1cd02c4d ("frontend/dri: move callbacks from the VTable into dri_screen, dri_drawable")
---
 src/gallium/frontends/dri/dri2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gallium/frontends/dri/dri2.c b/src/gallium/frontends/dri/dri2.c
index b66f9d69142e..df5041242851 100644
--- a/src/gallium/frontends/dri/dri2.c
+++ b/src/gallium/frontends/dri/dri2.c
@@ -2379,6 +2379,10 @@ dri_swrast_kms_init_screen(struct dri_screen *screen)
       screen->lookup_egl_image_validated = dri2_lookup_egl_image_validated;
    }
 
+   screen->create_drawable = dri2_create_drawable;
+   screen->allocate_buffer = dri2_allocate_buffer;
+   screen->release_buffer = dri2_release_buffer;
+
    return configs;
 
 destroy_screen:
-- 
GitLab

